docs/proposals/PROPOSAL_H.md

```diff
@@ -0,0 +1,352 @@
+      "oci.cpu.features": "AVX512FP16,",
+      "oci.kernel.selinux": "false",
+      "oci.kernel.configurations": "PREEMPT,",
+      "oci.kernel.versions": "[4.19, 5.16)",
+      "oci.os.glibc": "[2.31, 2.37]",
+      "oci.os.openmpi": "[4.1.6, 4.1.6]",
+      "oci.pci.devices": "15B3.020D,",
+      "oci.compatibility-score": 100,
+    },
+    {
+      "oci.cpu.vendor": "AuthenticAMD",
+      "oci.cpu.features": "FPHP,",
+      "oci.kernel.selinux": "false",
+      "oci.kernel.configurations": "PREEMPT,",
+      "oci.kernel.versions": "[4.19, 5.16)",
+      "oci.os.glibc": "[2.31, 2.37]",
+      "oci.os.openmpi": "[4.1.6, 4.1.6]",
+      "oci.pci.devices": "1002.67ff,",
+      "oci.compatibility-score": 80,
+      "annotation": "only work with AMD CPU and AMD GPU"
+    }
+  ]
+}
+```
+
+## Compatibilities Embeded Inside Platform Object
+
+### Extend Platform Object with New Compatibilities Field
+
+Add a new field in platform object in [image index](https://github.com/opencontainers/image-spec/blob/main/image-index.md#image-index-property-descriptions).
+
+- **`manifests`** _array of objects_
+
+  - **`platform`** _object_
+
+    This OPTIONAL property describes the minimum runtime requirements of the image.
+    This property SHOULD be present if its target is platform-specific.
+
+    - **`architecture`** _string_
+
+    - **`variant`** _string_
+
+    - **`os`** _string_
+
+    - **`os.version`** _string_
+
+    - **`os.features`** _array of strings_
+
+    - **`features`** _array of strings_
+
+    - **`compatibilities`** _list of map_
+
+      This new OPTIONAL property describes a list of compatibility sets, contains at least one compatibility set. Same definition as that in Image Compatibility Spec.
+
+### Migration and Deprecation
+
+The platform.os.version, platform.os.features and platform.features are expected to be migrated to platform.compatibilites fields.
+
+It may take years for the migration.
+
+The deprecation for these fields will start with pre-notification and migration transition period.
+
+Before all runtimes' migration finished, these fields should still be valid.
+
+## Flexibility for Different Purpose
+
+It's the image author's (or image provider's) freedom to use standalone artifact or platform.compatibilities to express the compatibility requirements, the format will be kept same.
+
+The subject field should be used to associate the compatibility artifact with target image. The referrers API should be used to discover artifacts.
+
+The client will query for an artifact using the referrers response, selecting the entry with a matching artifactType and the most recent "created" timestamp.
+
+If platform.compatibilities is detected and not empty, then runtime is not required to send referrs API to query compatibility artifact.
+
+If there is no platform.compatibilities, or the content of platform.compatibilities is  empty, then runtime should send referrs API to registry to query compatibility artifact.
```

Comment:
> then runtime should send referrs API to registry to query compatibility artifact.

it should be configurable whether the runtime or others to send referrers api request or not if  there is no platform.compatibilities field, or the content of it is empty